wgpu/backends/wgpu_native/_api.py

```diff
@@ -3550,27 +3551,50 @@ def write_texture(
             self._internal, c_destination, c_data, data_length, c_data_layout, c_size
         )
 
+    _shared_copy_buffer = None, 0
+
     def read_texture(self, source, data_layout, size):
         # Note that the bytes_per_row restriction does not apply for
         # this function; we have to deal with it.
 
         device = source["texture"]._device
 
         # Get and calculate striding info
+        # Note that full_stride (bytes per row) must be a multiple of 256
         ori_offset = data_layout.get("offset", 0)
         ori_stride = data_layout["bytes_per_row"]
         extra_stride = (256 - ori_stride % 256) % 256
         full_stride = ori_stride + extra_stride
 
         size = _tuple_from_extent3d(size)
+        data_length = full_stride * size[1] * size[2]
 
         # Create temporary buffer
-        data_length = full_stride * size[1] * size[2]
-        tmp_usage = flags.BufferUsage.COPY_DST | flags.BufferUsage.MAP_READ
-        tmp_buffer = device._create_buffer("", data_length, tmp_usage, False)
+        is_present_texture = source["texture"].label == "present"
+        copy_buffer = None
+        if is_present_texture:
+            copy_buffer, time_since_size_ok = self._shared_copy_buffer
+            if copy_buffer is None:
+                pass  # No buffer
+            elif copy_buffer.size < data_length:
+                copy_buffer = None  # Buffer too small
+            elif copy_buffer.size < data_length * 4:
+                self._shared_copy_buffer = copy_buffer, time.time()  # Bufer size ok
+            elif time.time() - time_since_size_ok > 5.0:
+                copy_buffer = None  # Too large too long
```

Comment:
@hmaarrfk this one. If this is a texture used for presenting, it checks the shared buffer. If it's large enough, and not too large for a too long time, it will re-use it.